### PR TITLE
Don't raise in sys_command_win32

### PR DESCRIPTION
**Pull request by hhugo**

### CODE REVIEW

**Comment by hhugo** [comment]
following https://github.com/ocaml/ocamlbuild/pull/339

File: src/my_std.ml [hunk 1]
```diff
@@ -347,11 +347,17 @@ let prepare_command_for_windows cmd =
...
+  try
+    let oc = Unix.open_process_args_out args.(0) args in
+    match Unix.close_process_out oc with
+    | WEXITED x -> x
+    | WSIGNALED _ -> 2 (* like OCaml's uncaught exceptions *)
+    | WSTOPPED _ -> 127
+  with (Unix.Unix_error _) as x ->
+    (* Sys.command doesn't raise an exception, so sys_command_win32 also won't
+       raise *)
+    Printexc.to_string x ^ ":" ^ cmd |> prerr_endline;
```
[the remark this answers is not in the document]
Comment (hhugo) [reply src/my_std.ml]: done

File: src/my_std.ml [hunk 2]
```diff
@@ -25,6 +25,12 @@ exception Exit_system_error of string
 exception Exit_with_code of int
 exception Exit_silently_with_code of int
 
+type log = { mutable dprintf : 'a. int -> ('a, Format.formatter, unit) format -> 'a }
+(* Here to break the circular dep *)
+let dprintf _lvl _fmt = failwith "My_std.log not initialized"
+let log = { dprintf = dprintf }
```
Comment (gasche) [inline src/my_std.ml:31]: I would define `dprintf` as a local variable of `log`, it is strange to add a broken `dprintf` function to the global scope of the module.

Comment (hhugo) [reply src/my_std.ml]: done

**Pull request merged**